posthog/clickhouse/property_groups.py

```diff
@@ -65,11 +101,25 @@ def get_alter_create_statements(self, name: str) -> Iterable[str]:
     "rdt_cid",  # reddit
 ]
 
-sharded_events_property_groups.register(
-    "custom",
-    PropertyGroupDefinition(
-        f"key NOT LIKE '$%' AND key NOT IN (" + f", ".join(f"'{name}'" for name in ignore_custom_properties) + f")"
-    ),
-)
+event_property_group_definitions = {
+    "properties": {
+        "custom": PropertyGroupDefinition(
+            f"key NOT LIKE '$%' AND key NOT IN (" + f", ".join(f"'{name}'" for name in ignore_custom_properties) + f")"
+        ),
+        "feature_flags": PropertyGroupDefinition("key like '$feature/%'"),
+    }
+}
 
-sharded_events_property_groups.register("feature_flags", PropertyGroupDefinition("key like '$feature/%'"))
+property_groups = PropertyGroupManager(
+    settings.CLICKHOUSE_CLUSTER,
+    {
+        "sharded_events": event_property_group_definitions,
+        "events": {
+            column_name: {
+                group_name: dataclasses.replace(group_definition, is_materialized=False)
```

Comment:
This is kind of clunky — maybe in retrospect it would have been better to have this as some sort of flag up at the table level rather than each individual property group definition…

Reply:
yeah it's definitely not easy reading